src/metrics.jl

```diff
@@ -882,24 +890,25 @@ function _validate_threshold_class(optimal_threshold_class, classes)
 end
 
 function per_class_confusion_statistics(predicted_soft_labels::AbstractMatrix,
-                                        elected_hard_labels::AbstractVector, thresholds)
+                                        elected_hard_labels::AbstractVector, thresholds;
+                                        harden_fn=harden_by_threshold)
     class_count = size(predicted_soft_labels, 2)
     return map(1:class_count) do i
         return per_threshold_confusion_statistics(predicted_soft_labels,
                                                   elected_hard_labels,
-                                                  thresholds, i)
+                                                  thresholds, i; harden_fn)
     end
 end
 
 function per_threshold_confusion_statistics(predicted_soft_labels::AbstractMatrix,
                                             elected_hard_labels::AbstractVector, thresholds,
-                                            class_index)
+                                            class_index; harden_fn=harden_by_threshold)
     confusions = [confusion_matrix(2) for _ in 1:length(thresholds)]
     for label_index in 1:length(elected_hard_labels)
         predicted_soft_label = predicted_soft_labels[label_index, class_index]
         elected = (elected_hard_labels[label_index] == class_index) + 1
         for (threshold_index, threshold) in enumerate(thresholds)
-            predicted = (predicted_soft_label >= threshold) + 1
+            predicted = harden_fn(predicted_soft_label, threshold) + 1
```

Comment:
entirely unrelated to this change, but this +1 is a little awkward! It makes sense though- going from binary in/out to class 1/2.

Reply:
yeah, agreed 100%

Reply:
added a comment